### PR TITLE
Implements `mt.block`

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?

Implement `mars.tensor.block`.

## Related issue number

Closes https://github.com/mars-project/mars/issues/569

### CODE REVIEW

**Comment by hekaisheng** [comment]
Could you please add some test cases with specific `chunk_size` to test the behavior with multiply chunks and improve the coverage?

**Comment by fyrestone** [comment]
> Could you please add some test cases with specific `chunk_size` to test the behavior with multiply chunks and improve the coverage?

OK. I will add some test cases.

File: mars/tensor/merge/tests/test_merge_execute.py [hunk 1]
```diff
@@ -222,6 +222,21 @@ def testBlockExecution(self):
                           [1., 1., 0., 0., 3.]])
         np.testing.assert_array_equal(r, expected)
 
+        # eye with different chunk sizes
+        a = eye(2, chunk_size=5) * 2
+        b = eye(3, chunk_size=7) * 3
```
Comment (hekaisheng) [inline mars/tensor/merge/tests/test_merge_execute.py:227]: when `chunk_size` is greater than tensor shape, the tensor still has one chunk.

Try to increase the shape like this `a = eye(9, chunk_size=5) * 2`

Comment (fyrestone) [reply mars/tensor/merge/tests/test_merge_execute.py:227]: Good suggestion. Thanks.

**Review by hekaisheng — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**